### ISSUE TITLE
Failed to run "pnpm run build".

### ISSUE THREAD

**Issue by LittleTerry** [haibingtown/segment-matting#12]
I was trying to run "pnpm run build", but it gave me erro as follows:
what should I do? 
I am using Ubuntu 22.04.
Many thanks!
![image](https://github.com/user-attachments/assets/08b4c05b-4fda-40f0-980e-3b8579404124)

**Comment by haibingtown** [comment]
Please check the version of onnxruntime-web using the following command:

`pnpm list onnxruntime-web`

**Comment by LittleTerry** [comment]
Hi, thanks for your response. Following what readme file exactly says, here are what I have done :

![image](https://github.com/user-attachments/assets/5a52c6f5-c19b-4ba1-a0dd-51e871f232c5)

![image](https://github.com/user-attachments/assets/f27734f6-dc29-4957-81ca-62bb7fbe3ed6)

![image](https://github.com/user-attachments/assets/23383bfb-ea87-46e0-9f2e-e1129e01586b)

![image](https://github.com/user-attachments/assets/6a7ab0ee-df40-43de-9d89-c187a0fdc625)

I did nothing but the onnxruntime-web versiong is 1.20.1

**Comment by haibingtown** [comment]
I have reproduced this issue, and it is caused by a version compatibility problem with onnxruntime-web. You can downgrade to version 1.14.0, and the issue will be resolved.

**Comment by LittleTerry** [comment]
> I have reproduced this issue, and it is caused by a version compatibility problem with onnxruntime-web. You can downgrade to version 1.14.0, and the issue will be resolved.

Thanks for your advice. I've downgraded the version of onnxruntime-web to 1.14.0

<img width="981" alt="image" src="https://github.com/user-attachments/assets/35e73e3b-dd0b-403e-bae9-737cfa0f1c10">

I ran "pnpm run build" again, but another erro happend. 

<img width="960" alt="image" src="https://github.com/user-attachments/assets/c298c868-25b1-4d10-9028-3fb12e6a418f">


I found the mentioned file is located here, but I've no idea what to do next. 

<img width="1210" alt="image" src="https://github.com/user-attachments/assets/e4356dc4-c38e-4dc6-b1cd-b0538077b176">

**Comment by haibingtown** [comment]
It seems to be an issue with the versions of some libraries after installing with pnpm. I was able to reproduce your problem, but switching to npm for install and run resolves the issue.

**Comment by LittleTerry** [comment]
> It seems to be an issue with the versions of some libraries after installing with pnpm. I was able to reproduce your problem, but switching to npm for install and run resolves the issue.

Hi, thanks for you advice.

I re-clone the project, then run "npm i".

<img width="987" alt="image" src="https://github.com/user-attachments/assets/f304f5b7-25b0-4c26-b79b-7b33b54fc266">

It gave me error as follow:

<img width="977" alt="image" src="https://github.com/user-attachments/assets/5f594723-bc83-4388-86ef-7258c69ab291">

Am I doing something wrong? Could you please update the read.me file? Many thanks!

**Comment by LittleTerry** [comment]
> It seems to be an issue with the versions of some libraries after installing with pnpm. I was able to reproduce your problem, but switching to npm for install and run resolves the issue.


Hi, thanks for you advice.

I re-cloned the project, then ran "sudo npm i", no error.

<img width="987" alt="image" src="https://github.com/user-attachments/assets/f304f5b7-25b0-4c26-b79b-7b33b54fc266">

After that, I ran "sudo npm run build", but it gave me the same error as follows:

<img width="974" alt="image" src="https://github.com/user-attachments/assets/4308bf8c-b963-466a-ad94-48a061e887bb">

I also tried to resolve the warning, then rerun "npm run build", but failed.
<img width="976" alt="image" src="https://github.com/user-attachments/assets/b32e8c03-1016-43a0-9ca3-509ef359faa9">

Was I doing something wrong? Could you please update the read.me file? Many thanks!

**Comment by chandrasuda** [comment]
@haibingtown I am facing the same issue now. @LittleTerry Were you able to solve this issue? Any advice would be greatly appreciated. 

<img width="864" alt="Image" src="https://github.com/user-attachments/assets/50d06cc7-f8ed-46b6-8e7e-cad0315b3ebc" />

If I comment out the Profiler, I am able to run the app, but the Segment Everything option isn't showing. Could you please suggest how to enable that? 

<img width="283" alt="Image" src="https://github.com/user-attachments/assets/2225759d-92e3-45de-9c06-4de98e4b48f4" />

**Comment by LittleTerry** [comment]
> [@haibingtown](https://github.com/haibingtown) I am facing the same issue now. [@LittleTerry](https://github.com/LittleTerry) Were you able to solve this issue? Any advice would be greatly appreciated.
> 
> <img alt="Image" width="864" src="https://private-user-images.githubusercontent.com/98287934/416041552-50d06cc7-f8ed-46b6-8e7e-cad0315b3ebc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAzNjczMjUsIm5iZiI6MTc0MDM2NzAyNSwicGF0aCI6Ii85ODI4NzkzNC80MTYwNDE1NTItNTBkMDZjYzctZjhlZC00NmI2LThlN2UtY2FkMDMxNWIzZWJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjI0VDAzMTcwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ1NGQzZWRlNjI5ZTQyMzQzZWNkODA0MzdlNTNkNDk0MDdjNmNlMThmZjg1Y2EwMGU0Y2U0N2U2NjQ5ODUwNWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.on_-Lu9J-aMI2U7b2xYsXNn87zW1ojW3lKeE3QE_KWg">
> If I comment out the Profiler, I am able to run the app, but the Segment Everything option isn't showing. Could you please suggest how to enable that?
> 
> <img alt="Image" width="283" src="https://private-user-images.githubusercontent.com/98287934/416042731-2225759d-92e3-45de-9c06-4de98e4b48f4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAzNjczMjUsIm5iZiI6MTc0MDM2NzAyNSwicGF0aCI6Ii85ODI4NzkzNC80MTYwNDI3MzEtMjIyNTc1OWQtOTJlMy00NWRlLTljMDYtNGRlOThlNGI0OGY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjI0VDAzMTcwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3OTI0ODNiYTAyZDZiMmRmMjI3YWU5M2M3NmFkNjM0NjkyNDBkMDBhM2VmYjMwN2Y2MDc3MzVkOWFhM2FjN2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.S-WpArvVJuIf6V0C_7_f5Q6QjdPBb-6vzy4Yv8AFgvA">

No, I didn't solve the problem. Still waiting for @haibingtown reply

**Comment by hiSandog** [comment]
Because multiple people are hitting this with both pnpm and npm, the project probably needs a locked, documented install path. Adding `packageManager`, an engines range, and a committed lockfile would make dependency resolution reproducible; then the build error can be debugged against one known dependency graph. If commenting out Profiler changes the UI behavior, that also suggests the README should mention whether the Profiler dependency is optional or required for “Segment Everything.”
